### ISSUE TITLE
Add option for Proxy Support

### ISSUE THREAD

**Issue by alec-hs** [aws-quickstart/quickstart-darktrace-vsensor#43]
Please could you add a configuration option to support the use of a proxy server such as SQUID. Best way would probably be to add some optional parameters:

- Proxy Address
- Proxy Port
- Comma separated list of domains/ips to ignore

Then have these added to /etc/environment of the instance used during AMI creation:

```
export http_proxy="http://proxyaddress.network:3128/"
export https_proxy="http://proxyaddress.network:3128/"
export ftp_proxy="http://proxyaddress.network:3128/"
export no_proxy="127.0.0.1,localhost,mydomain.com,amazonaws.com,169.254.169.254"

export HTTP_PROXY="http://proxyaddress.network:3128/"
export HTTPS_PROXY="http://proxyaddress.network:3128/"
export FTP_PROXY="http://proxyaddress.network:3128/"
export NO_PROXY="127.0.0.1,localhost,mydomain.com,amazonaws.com,169.254.169.254"
```

**Comment by alexdarktrace** [comment]
Hi Alec-hs, Thanks for the ticket. 
So that we can ensure we cover the use case, is this only for the normal vSensor connectivity:
1) for apt package updates to packages-cdn.darktrace.com
2) for connectivity to the master

Or are there more places where the proxy setting in /etc/environment would be helpful? Eg - are you running custom programs/scripts on the vSensor too?

**Comment by alec-hs** [comment]
Hi, this is just for use during the setup and normal connectivity - this would mean it would need to be set when the AMI is built too. In our case this would cover all apt updates (to Ubuntu packages and packages-cdn.darktrace.com). The connectivity to the master would not use the proxy hence the inclusion of the NO_PROXY option.

**Comment by alexdarktrace** [comment]
We already support setting the proxy for apt package updates and connectivity to the master, but these are not currently exposed to the quickstart as variables. They don't support NO_PROXY, but could be set independently. Would adding the ability to configure those satisfy your use cases?

**Comment by alec-hs** [comment]
Yep 100% that should be perfect. I guess it would be better to set proxy in the apt conf and also the /etc/environment to cover all bases. The reason for the NO_PROXY was because of the direct routing available for internal DNS records - ie the DT master node.
